### PR TITLE
Some Updates

### PR DESCRIPTION
**Pull request by jimmcgaw**
In this PR:

* I updated the `optimize()` call signature where it was invoked. This fixes an issue I introduced here: https://github.com/terranexum/OpenQGO/pull/6/files#diff-1d9f359b748b73580cae76b45d2e1e5768fa2f03a090f74cdf7ea2e39c367859R593
* Updated the README with a contributing section.
* Added some further type annotations to codebase.
* Added a requirements.txt file to explicitly specify dependencies.
* Added a .gitignore file so compiled bytecode and pycache dirs don't get added.
* Specify the Python version currently being used for pyenv. (.python-version file)

These are broken into separate commits for (hopefully) ease of review.

### CODE REVIEW

File: notebooks/geojson_handling_clear_creek_county.ipynb
Comment (jimmcgaw) [inline notebooks/geojson_handling_clear_creek_county.ipynb:1]: This looks huge, but all I did was update the `QGOOptimizer.optimize()` call to use the renamed parameter.

File: src/OpenQGO.py [hunk 1]
```diff
@@ -88,10 +88,16 @@
...
+
 # paths to input data in data_in
-links = ["/home/OpenQGO/data_in/mineral_production_facilities.geojson"]
-links.append("/home/OpenQGO/data_in/mineral_sources.geojson")
-links.append("/home/OpenQGO/data_in/possible_solar.geojson")
+links = [
+    DATA_IN_DIR / "mineral_production_facilities.geojson",
+    DATA_IN_DIR / "mineral_sources.geojson",
+    DATA_IN_DIR / "possible_solar.geojson",
+]
```
Comment (jimmcgaw) [inline src/OpenQGO.py:100]: This makes the file paths relative, so they'll work even outside of the docker container.

File: src/QGOOut.py [hunk 2]
```diff
@@ -65,16 +75,16 @@ def retNames(item):
...
 
         # Finally, generating the actual GeoDataFrame that can be manipulated
         geo_prob_df = gpd.GeoDataFrame(single_prob_df, geometry='geometry', crs='epsg:4326')
         geo_sol_df = gpd.GeoDataFrame(single_sol_df, geometry='geometry', crs='epsg:4326')
         
         # Output to file
-        geo_prob_df.to_file('/home/OpenQGO/data_out/QGOprob.geojson', driver='GeoJSON')  
-        geo_sol_df.to_file('/home/OpenQGO/data_out/QGOsol.geojson', driver='GeoJSON')  
+        geo_prob_df.to_file(DATA_OUT_DIR / 'QGOprob.geojson', driver='GeoJSON')  
+        geo_sol_df.to_file(DATA_OUT_DIR / 'QGOsol.geojson', driver='GeoJSON')
```
Comment (jimmcgaw) [inline src/QGOOut.py:87]: This makes the file paths relative, so they'll work even outside of the docker container.

**Review by dahlwinters — APPROVED** [review]
Solid updates that significantly improve the quality of our codebase, approved!

**Pull request merged**